<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>750</width>
    <height>652</height>
   </rect>
  </property>
  <property name="font">
   <font>
    <pointsize>14</pointsize>
   </font>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QStackedWidget" name="playingPage">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>20</y>
      <width>711</width>
      <height>531</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>1</number>
    </property>
    <widget class="QWidget" name="page">
     <widget class="QWidget" name="mainWidget" native="true">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>60</y>
        <width>681</width>
        <height>321</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QWidget" name="gameStateWidget" native="true">
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>GameState : </string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="gameStateLabel">
            <property name="text">
             <string>Standby</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QWidget" name="loadTypeWidget" native="true">
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>載入類型 : </string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="boardTypeComboBox">
            <item>
             <property name="text">
              <string>盤面檔</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>指定炸彈數量盤面</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>指定炸彈升成率盤面</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QStackedWidget" name="stackedWidget">
         <property name="currentIndex">
          <number>0</number>
         </property>
         <widget class="QWidget" name="boardFileWidgetPage">
          <widget class="QWidget" name="boardFileWidget" native="true">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>10</y>
             <width>679</width>
             <height>73</height>
            </rect>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout_8">
            <item>
             <widget class="QLabel" name="label">
              <property name="text">
               <string>盤面檔位置 : </string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QTextEdit" name="filePathTextEdit"/>
            </item>
            <item>
             <widget class="QPushButton" name="Load1Button">
              <property name="text">
               <string>Load</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
         <widget class="QWidget" name="bombCountWidgetPage">
          <widget class="QWidget" name="bombCountWidget" native="true">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>10</y>
             <width>679</width>
             <height>51</height>
            </rect>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout_7">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_4">
              <item>
               <widget class="QLabel" name="label_5">
                <property name="text">
                 <string>Rows : </string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSpinBox" name="rowsAmountSpinBox1">
                <property name="minimum">
                 <number>5</number>
                </property>
                <property name="maximum">
                 <number>200</number>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_5">
              <item>
               <widget class="QLabel" name="label_6">
                <property name="text">
                 <string>Cols : </string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSpinBox" name="colsAmountSpinBox1">
                <property name="minimum">
                 <number>5</number>
                </property>
                <property name="maximum">
                 <number>200</number>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_6">
              <item>
               <widget class="QLabel" name="label_7">
                <property name="text">
                 <string>炸彈數量 : </string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSpinBox" name="bombAmountSpinBox">
                <property name="maximum">
                 <number>200</number>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="load2Button">
                <property name="text">
                 <string>Load</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </widget>
         <widget class="QWidget" name="bombRateWidgetPage">
          <widget class="QWidget" name="bombRateWidget" native="true">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>10</y>
             <width>679</width>
             <height>65</height>
            </rect>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout_10">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_11">
              <item>
               <widget class="QLabel" name="label_8">
                <property name="text">
                 <string>Rows : </string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSpinBox" name="rowsAmountSpinBox2"/>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_12">
              <item>
               <widget class="QLabel" name="label_9">
                <property name="text">
                 <string>Cols : </string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSpinBox" name="colsAmountSpinBox2"/>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_13">
              <item>
               <widget class="QLabel" name="label_10">
                <property name="text">
                 <string>炸彈生成機率 : </string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QDoubleSpinBox" name="bombRateSpinBox">
                <property name="singleStep">
                 <double>0.020000000000000</double>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="load3Button">
                <property name="text">
                 <string>Load</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </widget>
        </widget>
       </item>
       <item>
        <widget class="QWidget" name="printWidget" native="true">
         <layout class="QHBoxLayout" name="horizontalLayout_9">
          <item>
           <widget class="QPushButton" name="PrintGameBoardButton1">
            <property name="text">
             <string>PrintGameBoard</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="PrintGameAnswerButton1">
            <property name="text">
             <string>PrintGameAnswer</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="PrintGameStateButton1">
            <property name="text">
             <string>PrintGameState</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="startGameButton">
         <property name="text">
          <string>StartGame</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QWidget" name="page_2">
     <property name="font">
      <font>
       <pointsize>10</pointsize>
      </font>
     </property>
     <widget class="QWidget" name="verticalLayoutWidget">
      <property name="geometry">
       <rect>
        <x>13</x>
        <y>15</y>
        <width>681</width>
        <height>161</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_14">
         <item>
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>GameStatus : </string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="gameState1Label">
           <property name="text">
            <string>Playing</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_17" stretch="0">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_15">
           <item>
            <widget class="QLabel" name="label_12">
             <property name="text">
              <string>Bomb Count :</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="bombAmountLabel">
             <property name="text">
              <string/>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_16">
             <item>
              <widget class="QLabel" name="label_13">
               <property name="text">
                <string>Flag Count :</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="flagAmountLabel">
               <property name="text">
                <string/>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_18">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_19">
           <item>
            <widget class="QLabel" name="label_14">
             <property name="text">
              <string>Open Blank Count :</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="openBlankAmountLabel">
             <property name="text">
              <string/>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_20">
             <item>
              <widget class="QLabel" name="label_15">
               <property name="text">
                <string>Remain Blank Count : </string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="remainBlankAmountLabel">
               <property name="text">
                <string/>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QWidget" name="printWidget_2" native="true">
         <layout class="QHBoxLayout" name="horizontalLayout_22">
          <item>
           <widget class="QPushButton" name="PrintGameBoardButton2">
            <property name="text">
             <string>PrintGameBoard</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="PrintGameAnswerButton2">
            <property name="text">
             <string>PrintGameAnswer</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="PrintGameStateButton2">
            <property name="text">
             <string>PrintGameState</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QScrollArea" name="scrollArea">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>190</y>
        <width>681</width>
        <height>341</height>
       </rect>
      </property>
      <property name="verticalScrollBarPolicy">
       <enum>Qt::ScrollBarAlwaysOn</enum>
      </property>
      <property name="horizontalScrollBarPolicy">
       <enum>Qt::ScrollBarAlwaysOn</enum>
      </property>
      <property name="sizeAdjustPolicy">
       <enum>QAbstractScrollArea::AdjustIgnored</enum>
      </property>
      <property name="widgetResizable">
       <bool>true</bool>
      </property>
      <widget class="QWidget" name="scrollAreaWidgetContents">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>662</width>
         <height>322</height>
        </rect>
       </property>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="page_3">
     <widget class="QPushButton" name="replayButton">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>210</y>
        <width>151</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Replay</string>
      </property>
     </widget>
     <widget class="QPushButton" name="quitButton">
      <property name="geometry">
       <rect>
        <x>380</x>
        <y>210</y>
        <width>151</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Quit</string>
      </property>
     </widget>
     <widget class="QWidget" name="horizontalLayoutWidget">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>50</y>
        <width>481</width>
        <height>41</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QLabel" name="label_11">
         <property name="text">
          <string>GameStatus : </string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_16">
         <property name="text">
          <string>GameOver</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </widget>
   <widget class="QWidget" name="boardWidget" native="true">
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>540</y>
      <width>160</width>
      <height>80</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="boardLayout"/>
   </widget>
   <widget class="QSlider" name="musiclSlider">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>10</y>
      <width>160</width>
      <height>18</height>
     </rect>
    </property>
    <property name="maximum">
     <number>100</number>
    </property>
    <property name="value">
     <number>25</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>750</width>
     <height>35</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
